crates/matrix-sdk-crypto/src/dehydrated_devices.rs

```diff
@@ -318,6 +318,7 @@ impl DehydratedDevice {
         let mut transaction = self.store.transaction().await;
 
         let account = transaction.account().await?;
+        account.mark_as_dehydrated();
```

Comment:
This should happen in the `DehydratedDevices::create()` method.

Reply:
Yup.  I just put this here for now because I didn't want to touch the storage parts yet (which is also why I didn't put it in a constructor).